File: ClientRuntimes/NodeJS/lib/clientRuntime.js
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

var exports = module.exports;

var utils = require('./util/util');

var nodeVersion = utils.getNodeVersion();
if (nodeVersion.major === 0 && nodeVersion.minor > 8 && !(nodeVersion.minor > 10 || (nodeVersion.minor === 10 && nodeVersion.patch >= 3))) {
  throw new Error('The Microsoft client runtime does not work with node versions > 0.8.22 and < 0.10.3, due to security issues. Please upgrade to node >= 0.10.3');
}

exports.Constants = require('./util/constants');
exports.Logger = require('./diagnostics/logger');

exports.WebResource = require('./http/webResource');
exports.ServiceClient = require('./services/serviceClient');

// Credentials
exports.TokenCredentials = require('./services/credentials/tokenCredentials');
exports.BasicAuthenticationCredentials = require('./services/credentials/basicAuthenticationCredentials');

// Other filters
exports.LinearRetryPolicyFilter = require('./services/filters/linearRetryPolicyFilter');
exports.ExponentialRetryPolicyFilter = require('./services/filters/exponentialRetryPolicyFilter');
exports.ProxyFilter = require('./services/filters/proxyFilter');
exports.LogFilter = require('./services/filters/logFilter');
exports.SigningFilter = require('./services/filters/signingFilter');

exports.validate = require('./util/validate');
exports.requestPipeline = require('./http/requestPipeline');

File: ClientRuntimes/Ruby/ms-rest/lib/client_runtime.rb
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License. See License.txt in the project root for license information.

require 'Base64'
require 'openssl'
require 'net/http'
require 'client_runtime/version'

require 'client_runtime/credentials/service_client_credentials'
require 'client_runtime/credentials/basic_authentication_credentials'
require 'client_runtime/credentials/token_credentials'

require 'client_runtime/deserialization_error.rb'
require 'client_runtime/serialization.rb'
require 'client_runtime/http_operation_response'
require 'client_runtime/http_operation_exception'
require 'client_runtime/service_client'

module ClientRuntime; end
